File: docs/snippets/all/archetypes/image_encoded.py
"""Create and log an image."""

from pathlib import Path

import rerun as rr

image_file_path = Path(__file__).parent / "ferris.png"

rr.init("rerun_example_image_encoded", spawn=True)

rr.log("image", rr.ImageEncoded(path=image_file_path))

File: docs/snippets/all/archetypes/image_encoded.cpp
// Create and log a image.

#include <rerun.hpp>

#include <filesystem>
#include <fstream>
#include <iostream>
#include <vector>

namespace fs = std::filesystem;

int main() {
    const auto rec = rerun::RecordingStream("rerun_example_image_encoded");
    rec.spawn().exit_on_failure();

    fs::path image_filepath = fs::path(__FILE__).parent_path() / "ferris.png";

    rec.log("image", rerun::ImageEncoded::from_file(image_filepath).value_or_throw());
}
